cpp/src/randomforest/randomforest.cu

```diff
@@ -341,6 +347,102 @@ std::vector<unsigned char> save_model(ModelHandle model) {
   return bytes_info;
 }
 
+void check_concat_tl_mod(ModelHandle concat_tree_handle,
+                         std::vector<ModelHandle*> treelite_handles) {
+  size_t concat_forest;
+  size_t single_handle;
+  int total_num_trees;
+  for (int tl_num = 0; tl_num < treelite_handles.size(); tl_num++) {
+    TREELITE_CHECK(
+      TreeliteQueryNumTree(*treelite_handles[tl_num], &single_handle));
+    total_num_trees = total_num_trees + single_handle;
+  }
+
+  TREELITE_CHECK(TreeliteQueryNumTree(concat_tree_handle, &concat_forest));
+
+  ASSERT(
+    concat_forest != total_num_trees,
+    "Error! the number of trees in the concatenated forest and the sum "
+    "of the trees present in the forests present in each worker are not equal");
+
+  int concat_mod_tree_num = 0;
+  tl::Model& concat_model = *(tl::Model*)concat_tree_handle;
+  for (int tl_num = 0; tl_num < treelite_handles.size(); tl_num++) {
+    const tl::Model& model = *(tl::Model*)(treelite_handles[tl_num]);
+    bool check_num_features = concat_model.num_feature != model.num_feature;
```

Comment:
I would probably just do `ASSERT(concat_model.num_feature == model.num_feature, "Feature coutn mismatch")` and so on. Then you get a more specific error message and don't have to manage these locals.